pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -106,141 +173,263 @@ func (c *Compactor) stopping(_ error) error {
-func NoopGetSeries() *v1.Series { return nil }
+	return chunkRefs
+}
 
-// TODO Then get chunk data from series
-func NoopGetChunks() []byte { return nil }
+// TODO Revisit this step once v1/bloom lib updated to combine blooms in the same series
+func buildBloomBlock(bloomForChks v1.SeriesWithBloom, series Series) (bloomshipper.Block, error) {
+	localDst := createLocalFileName(series)
 
-// part1: Create a compact method that assumes no block/meta files exists (eg first compaction)
-// part2: Write logic to check first for existing block/meta files and does above.
-func (c *Compactor) compactNewChunks(ctx context.Context, dst string) (err error) {
-	// part1
-	series := NoopGetSeries()
-	data := NoopGetChunks()
+	//write bloom to a local dir
+	builder, err := v1.NewBlockBuilder(v1.NewBlockOptions(), v1.NewDirectoryBlockWriter(localDst))
+	if err != nil {
+		level.Info(util_log.Logger).Log("creating builder", err)
+		return bloomshipper.Block{}, err
+	}
 
-	bloom := v1.Bloom{ScalableBloomFilter: *filter.NewDefaultScalableBloomFilter(0.01)}
-	// create bloom filters from that.
-	bloom.Add([]byte(fmt.Sprint(data)))
+	err = builder.BuildFrom(v1.NewSliceIter([]v1.SeriesWithBloom{bloomForChks}))
+	if err != nil {
+		level.Info(util_log.Logger).Log("writing bloom", err)
+		return bloomshipper.Block{}, err
+	}
 
-	// block and seriesList
-	seriesList := []v1.SeriesWithBloom{
-		{
-			Series: series,
-			Bloom:  &bloom,
-		},
+	blockFile, err := os.Open(filepath.Join(localDst, "bloom"))
+	if err != nil {
+		level.Info(util_log.Logger).Log("reading bloomBlock", err)
 	}
 
-	writer := v1.NewDirectoryBlockWriter(dst)
+	// read the checksum
+	if _, err := blockFile.Seek(-4, 2); err != nil {
+		return bloomshipper.Block{}, errors.Wrap(err, "seeking to bloom checksum")
+	}
+	checksum := make([]byte, 4)
+	if _, err := blockFile.Read(checksum); err != nil {
+		return bloomshipper.Block{}, errors.Wrap(err, "reading bloom checksum")
+	}
 
-	builder, err := v1.NewBlockBuilder(
-		v1.BlockOptions{
-			SeriesPageSize: 100,
-			BloomPageSize:  10 << 10,
-		}, writer)
-	if err != nil {
-		return err
+	// Reset back to beginning
+	if _, err := blockFile.Seek(0, 0); err != nil {
+		return bloomshipper.Block{}, errors.Wrap(err, "seeking to back to beginning of the file")
+	}
+
+	objectStoreDst := createObjStorageFileName(series, fmt.Sprint(binary.BigEndian.Uint32(checksum)))
+
+	blocks := bloomshipper.Block{
+		BlockRef: bloomshipper.BlockRef{
+			Ref: bloomshipper.Ref{
+				TenantID:       series.tenant,
+				TableName:      series.tableName,
+				MinFingerprint: uint64(series.fingerPrint), //TODO will change once we compact multiple blooms into a block
+				MaxFingerprint: uint64(series.fingerPrint),
+				StartTimestamp: series.from.Unix(),
+				EndTimestamp:   series.through.Unix(),
+				Checksum:       binary.BigEndian.Uint32(checksum),
+			},
+			IndexPath: series.indexPath, // For reviewer, this is the same filepath with the original TSDBindex for this series?
```

Comment:
yes, it's the path to the index file from which the block is built, to understand that we need to rebuild the block file if the index file is changed.